balancer/endpointsharding/endpointsharding.go

```diff
@@ -150,9 +172,10 @@ func (es *endpointSharding) UpdateClientConnState(state balancer.ClientConnState
 	// Delete old children that are no longer present.
 	for _, e := range children.Keys() {
 		child, _ := children.Get(e)
-		bal := child.(balancer.Balancer)
+		bal := child.(*balancerWrapper)
 		if _, ok := newChildren.Get(e); !ok {
 			bal.Close()
+			bal.isClosed = true
```

Comment:
Can we move this into `balancerWrapper`'s `Close` method?

Also since calls into a `balancer.Balancer` must not happen concurrently, I don't think it requires a lock.

Reply:
Added a Close method and moved this into the Close method.